lib/OpenQA/Shared/Controller/Running.pm

```diff
@@ -113,8 +113,8 @@ sub streamtext ($self, $file_name, $start_hook = undef, $close_hook = undef) {
     # Setup utility function to close the connection if something goes wrong
     my $timer_id;
     my $close_connection = sub ($self) {
+        log_debug "Connection to $worker will be terminated because $logfile changed";
         Mojo::IOLoop->remove($timer_id);
-        $close_hook->();
```

Comment:
Why do you remove this? If you remove this because it is called via `$self->on(finish => …` anyway then `Mojo::IOLoop->remove($timer_id);` should *perhaps* be removed as well. It would make sense to think this though and state the reasoning in the commit message. Otherwise I'd refrain from removing this line.

If the close hook cannot handle the situation of sometimes being called with 2 arguments and sometimes being called with 0 arguments than the hook should be fixed or we always call it with 2 arguments.